code/modules/overmap/ftl_shunt/computer.dm

```diff
@@ -0,0 +1,150 @@
+
+	data["ftlstatus"] = linked_core.get_status()
+	data["shunt_x"] = linked_core.shunt_x
+	data["shunt_y"] = linked_core.shunt_y
+	data["fuel_joules"] = (linked_core.get_fuel(linked_core.fuel_ports) / 1000)
+	data["jumpcost"] = recalc_cost()
+	data["chargetime"] = round((linked_core.get_charge_time() / 600))
+	data["chargepercent"] = linked_core.chargepercent
+	data["maxfuel"] = linked_core.get_fuel_maximum(linked_core.fuel_ports)
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "ftl_computer.tmpl", "[linked.name] Superluminal Shunt Control", 420, 530, src)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/computer/ship/ftl/OnTopic(var/mob/user, var/list/href_list, state)
+	if(..())
+		return ..()
+
+	if (!linked)
+		return TOPIC_NOACTION
+
+	if(href_list["set_shunt_x"])
+		if(linked_core)
+			var/fumble = 0
+			if(!user.skill_check(SKILL_PILOT, SKILL_ADEPT))
+				fumble = rand(-2, 2)
+			var/input_x = input(user, "Enter Destination X Coordinates", "FTL Computer", linked_core.shunt_x) as num|null
+			if(input_x >= GLOB.using_map.overmap_size)
+				input_x = min(input_x, GLOB.using_map.overmap_size - 1)
+			if(CanInteract(user, state))
+				if(isnull(input_x))
+					input_x = linked_core.shunt_x
+				linked_core.shunt_x = clamp((input_x + fumble), 1, (GLOB.using_map.overmap_size - 1))
+				if(fumble)
+					to_chat(user, SPAN_WARNING("You fumble the input because of your inexperience!"))
+			return TOPIC_REFRESH
+
+	if(href_list["set_shunt_y"])
+		if(linked_core)
+			var/fumble = 0
+			if(!user.skill_check(SKILL_PILOT, SKILL_ADEPT))
+				fumble = rand(-2, 2)
+			var/input_y = input(user, "Enter Destination Y Coordinates", "FTL Computer", linked_core.shunt_y) as num|null
+			if(input_y >= GLOB.using_map.overmap_size)
+				input_y = min(input_y, GLOB.using_map.overmap_size - 1)
+			if(CanInteract(user, state))
+				if(isnull(input_y))
+					input_y = linked_core.shunt_y
+				linked_core.shunt_y = clamp((input_y + fumble), 1, (GLOB.using_map.overmap_size - 1))
+				if(fumble)
+					to_chat(user, SPAN_WARNING("You fumble the input because of your inexperience!"))
+			return TOPIC_REFRESH
+
+	if(href_list["start_shunt"])
+		if(linked_core)
+			if(linked_core.charging)
+				to_chat(user, SPAN_NOTICE("Superluminal jump charge in progress. Please wait for completion of jump!"))
+				return TOPIC_REFRESH
+			if(linked_core.get_status() != FTL_STATUS_GOOD)
+				to_chat(user, SPAN_WARNING("Superluminal shunt inoperable. Please try again later."))
+				return TOPIC_REFRESH
+
+			var/dist = get_dist(locate(linked_core.shunt_x, linked_core.shunt_y, GLOB.using_map.overmap_z), get_turf(linked))
+			if(dist > 3) //We are above the safe jump distance, give them a warning.
+				var/warning = alert(user, "Current shunt distance is above safe distance! Do you wish to continue?","Jump Safety System", "Yes", "No")
+				if(warning == "No" && CanInteract(user, state))
```

Comment:
Shouldn't this be `|| !CanInteract(user, state))`?

Reply:
No, this returns early if you say you don't want to proceed with the jump. The interaction check presumably returns TRUE/FALSE, so.